server/routes/createGoal/createGoalController.ts

```diff
@@ -107,10 +111,10 @@ export default class CreateGoalController {
   submitAddNoteForm: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
     const { createGoalForm } = req.session
-    const { addStepForm } = req.session
+    const { addStepForms } = req.session
     req.session.addNoteForm = { ...req.body }
 
-    const createGoalDto = toCreateGoalDto(createGoalForm, addStepForm, req.body)
+    const createGoalDto = toCreateGoalDto(createGoalForm, addStepForms, req.body)
     await this.educationAndWorkPlanService.createGoal(createGoalDto, req.user.token)
 
```

Comment:
I think we should nuke the forms in the session here as well - ie. the CreateGoalForm, the AddStepForm, the collection of AddStepForms; and the AddNoteForm

If we dont it means you can go back to create a new goal , perhaps for a new prisoner, and all the goal data from prisoner 1 will still be there! 🤣